lib/convertUnits.ts

```diff
@@ -0,0 +1,32 @@
+import { BigNumber } from 'bignumber.js'
+
+import { tokenPrecision, TokenUnitType, unitPrecision, UnitType } from './convertProperties'
+
+BigNumber.set({ EXPONENTIAL_AT: 1000 })
+
+export function convertUnit<T extends string>(
+  value: string,
+  fromUnit: T,
+  toUnit: T,
+  precisionDict: PrecisionDict<T>,
+): string | undefined {
+  return new BigNumber(value)
+    .times(unitPrecisionToValue(precisionDict[fromUnit]))
+    .div(unitPrecisionToValue(precisionDict[toUnit]))
+    .toString()
+}
+
+export function convertEthUnits(value: string, fromUnit: UnitType, toUnit: UnitType): string | undefined {
```

Comment:
I'd move `convertEthUnits` and `convertTokenUnits` to the beginning of the file as these functions are public interface of the module. @hasparus thoughts?